x/pool-incentives/keeper/grpc_query.go

```diff
@@ -157,6 +158,45 @@ func (q Querier) IncentivizedPools(ctx context.Context, _ *types.QueryIncentiviz
 		}
 	}
 
+	// Retrieve the migration records between balancer pools and concentrated liquidity pools.
+	// This comes from the superfluid keeper, since superfluid is the only pool incentives connected
+	// module that has access to the gamm modules store.
+	migrationRecords, err := q.superfluidKeeper.GetAllMigrationInfo(sdkCtx)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	// Iterate over all migration records.
+	for _, record := range migrationRecords.BalancerToConcentratedPoolLinks {
+		linkedClPoolIsIncentivized := false
+		var duration time.Duration
+
+		// If the linked concentrated liquidity pool is in the list of incentivized pools,
+		// note this and add its balancer counterpart to the list of incentivized pools.
+		for _, incentivizedPool := range incentivizedPools {
+			if incentivizedPool.PoolId == record.ClPoolId {
+				linkedClPoolIsIncentivized = true
+				duration = incentivizedPool.LockableDuration
```

Comment:
Is it correct to use this for duration?

AFAIR, we update the largest lockable duration gauge for the balancer pool while using incentivized epoch duration for CL pool which might not necessarily match

Reply:
Its just a holder value, but we can set this as the longest duration just in case, because I agree that is "default"